### PR TITLE
feat(modal): add option for fixed-height scrollable modal

### PR DESCRIPTION
**Pull request by jacksongonzales**
# Pull Request Template

## Issue Overview

This PR addresses [DAH-1157](https://sfgovdt.jira.com/jira/software/c/projects/DAH/boards/122?modal=detail&selectedIssue=DAH-1157)

## Description

We need to approximate the appearance of the lottery results modal on the angular version of DAHLIA. This PR adds an optional prop `scrollable` that makes the modal a fixed height and scrollable for long body content.

## How Can This Be Tested/Reviewed?

Run storybook and open the scrollable modal story

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [ ] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [ ] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: ui-components/src/overlays/Modal.tsx [hunk 1]
```diff
@@ -49,7 +52,7 @@ export const Modal = (props: ModalProps) => {
       <div className="modal">
         <ModalHeader title={props.title} uniqueId={uniqueIdRef.current} />
 
-        <section className="modal__inner">
+        <section className={classNames.join(" ")}>
```
Comment (christine-sfg) [inline ui-components/src/overlays/Modal.tsx:55]: would it be helpful to add classnames as a prop or is it unnecessary?

Comment (christine-sfg) [inline ui-components/src/overlays/Modal.tsx:55]: would it be helpful to add classnames as a prop or is it unnecessary?

File: ui-components/src/overlays/Modal.scss [hunk 2]
```diff
@@ -34,6 +34,11 @@
   &:last-of-type {
     @apply rounded-b;
   }
+
+  &.is-scrollable {
+    height: 80vh;
```
Comment (emilyjablonski) [inline ui-components/src/overlays/Modal.scss:39]: Screen Shot 2022-05-31 at 5 27 26 PM
This style can result in the footer being cut off at small screen sizes. Some sort of calc based on the view height I think will prevent this, something like `max-height: calc(100vh - 200px)` - let me know what you think!

**Review by emilyjablonski — CHANGES_REQUESTED** [review]
I like the new prop and story! I've got one comment on some height styling, and then I think as part of the DAH ticket we need the ability to add some class names to allow for reduced padding / margin. Let me know if you'd like to do this in a separate ticket.

**Review by christine-sfg — APPROVED** [review]

**Review by emilyjablonski — APPROVED** [review]

**Pull request merged**